src/GraphQL.MemoryCache/AutomaticPersistedQueries/AutomaticPersistedQueriesExecutionBase.cs

```diff
@@ -0,0 +1,132 @@
+using System.Security.Cryptography;
+using System.Text;
+using GraphQL.DI;
+
+namespace GraphQL.Caching;
+
+/// <summary>
+/// Implementation of Automatic Persisted Queries (APQ).
+/// https://www.apollographql.com/docs/react/api/link/persisted-queries/
+/// </summary>
+public abstract class AutomaticPersistedQueriesExecutionBase : IConfigureExecution
+{
+    /// <summary>
+    /// Supported version.
+    /// </summary>
+    public const string SUPPORTED_VERSION = "1";
+
+    private readonly object _lockObject = new();
+    private readonly SHA256 _sha256 = SHA256.Create();
+
+    /// <summary>
+    /// Searching APQ fields in <see cref="ExecutionOptions.Extensions"/> based on a protocol:
+    /// https://www.apollographql.com/docs/react/api/link/persisted-queries/#protocol
+    /// </summary>
+    public virtual (string? Hash, string? Version, bool Enabled) GetAPQProperties(Inputs? extensions)
+    {
+        string? hashResult = null;
+        string? versionResult = null;
+        bool enabledResult = false;
+
+        if (extensions?.TryGetValue("persistedQuery", out var persistedQueryObject) ?? false)
+        {
+            enabledResult = true;
+
+            if (persistedQueryObject is Dictionary<string, object> persistedQuery)
+            {
+                if (persistedQuery.TryGetValue("sha256Hash", out var sha256HashObject) && sha256HashObject is string sha256Hash && !string.IsNullOrEmpty(sha256Hash))
+                {
+                    hashResult = sha256Hash;
+                }
+
+                if (persistedQuery.TryGetValue("version", out var versionObject) && versionObject is string version && !string.IsNullOrEmpty(version))
+                {
+                    versionResult = version;
+                }
+            }
+        }
+
+        return (hashResult, versionResult, enabledResult);
+    }
+
+    /// <summary>
+    /// Compute SHA256 hash.
+    /// </summary>
+    public virtual string ComputeSHA256(string input)
+    {
+        byte[] bytes;
+        lock (_lockObject)
+        {
+            bytes = _sha256.ComputeHash(Encoding.UTF8.GetBytes(input));
+        }
+
+        var builder = new StringBuilder();
+        foreach (var item in bytes)
+        {
+            builder.Append(item.ToString("x2"));
+        }
+
+        return builder.ToString();
```

Comment:
I would check and see if this works properly and include it if so.

```suggestion
#if NET5
        return Convert.ToHexString(bytes);
#else
        var builder = new StringBuilder();
        foreach (var item in bytes)
        {
            builder.Append(item.ToString("x2"));
        }

        return builder.ToString();
#endif
```

Reply:
Note that ToHexString encodes as uppercase...

Reply:
Only `<TargetFramework>netstandard2.0</TargetFramework>` [available](https://github.com/graphql-dotnet/graphql-dotnet/blob/5.2.0/src/GraphQL.MemoryCache/GraphQL.MemoryCache.csproj#L4)

Reply:
Just add net5 tfm. For example, main package declares supported tfms like that
`  <TargetFrameworks>netstandard2.1;netstandard2.0;net6</TargetFrameworks>`

Reply:
resolved